tests/Expectation.lua

```diff
@@ -60,6 +60,70 @@ return {
             ("Error message does not match:\n%s\n"):format(message)
         )
     end,
+    ["it should fail if an empty function is expected to throw with a message"] = function()
+        local function shouldNotThrow()
+        end
+
+        local expect = Expectation.new(shouldNotThrow)
+
+        local success, message = pcall(function()
+            expect:throw("foo")
+        end)
+
+        assert(not success, "should fail")
+        assert(
+            message:match("Expected function to throw an error containing \"foo\", but it did not."),
+            ("Error message does not match:\n%s\n"):format(message)
+        )
+    end,
+    ["it should fail if a throwing function is expected to throw with a different error message"] = function()
+        local function shouldThrow()
+            error("oof")
+        end
+
+        local expect = Expectation.new(shouldThrow)
+
+        local success, message = pcall(function()
+            expect:throw("foo")
+        end)
+
+        assert(not success, "should fail")
+        assert(
+            message:match("Expected function to throw an error containing \"foo\", but it threw: .+%.lua:%d+: oof"),
+            ("Error message does not match:\n%s\n"):format(message)
+        )
+    end,
+    ["it should succeed if a throwing function is expected to throw a substring of the message"] = function()
+        local function shouldThrow()
+            error("foo-oof")
+        end
+
+        local expect = Expectation.new(shouldThrow)
+
+        local success = pcall(function()
+            expect:throw("oof")
+        end)
+
+        assert(success, "should succeed")
+    end,
+    ["it should fail if a throwing function is expected to never throw a substring of the message"] = function()
```

Comment:
For documentation purposes, it may be worth including "should succeed if it doesn't fail when it was expecting to never throw with a message". That is, for an empty function, expect.never:throw("foo") should succeed.